src/main/java/org/opensearch/knn/index/query/KNNWeight.java

```diff
@@ -441,14 +445,72 @@ private boolean isExactSearchThresholdSettingSet(int filterThresholdValue) {
         return filterThresholdValue != KNNSettings.ADVANCED_FILTERED_EXACT_SEARCH_THRESHOLD_DEFAULT_VALUE;
     }
 
+    /**
+     * This condition mainly checks whether exact search should be performed or not
+     * @param context LeafReaderContext
+     * @param filterIdsCount count of filtered Doc ids
+     * @param annResultCount Count of Nearest Neighbours we got after doing filtered ANN Search.
+     * @return boolean - true if exactSearch needs to be done after ANNSearch.
+     */
+    private boolean canDoExactSearch(final LeafReaderContext context, final int filterIdsCount, final int annResultCount) {
+        if (annResultCount == 0 && isMissingNativeEngineFiles(context)) {
```

Comment:
why we are looking for `annResultCount == 0` ?

Reply:
To narrow the calls to checkNativeEngine files are missing or not? For ex: if result != 0, we know that it is not missing.